src/Core/MySQL/PacketsProtocolText.cpp

```diff
@@ -173,30 +185,52 @@ ColumnDefinition getColumnDefinition(const String & column_name, const TypeIndex
         case TypeIndex::Float32:
             column_type = ColumnType::MYSQL_TYPE_FLOAT;
             flags = ColumnDefinitionFlags::BINARY_FLAG;
+            decimals = 31;
             break;
         case TypeIndex::Float64:
             column_type = ColumnType::MYSQL_TYPE_DOUBLE;
             flags = ColumnDefinitionFlags::BINARY_FLAG;
+            decimals = 31;
             break;
         case TypeIndex::Date:
+        case TypeIndex::Date32:
             column_type = ColumnType::MYSQL_TYPE_DATE;
             flags = ColumnDefinitionFlags::BINARY_FLAG;
             break;
         case TypeIndex::DateTime:
             column_type = ColumnType::MYSQL_TYPE_DATETIME;
             flags = ColumnDefinitionFlags::BINARY_FLAG;
             break;
-        case TypeIndex::String:
-        case TypeIndex::FixedString:
-            column_type = ColumnType::MYSQL_TYPE_STRING;
-            charset = CharacterSet::utf8_general_ci;
+        case TypeIndex::DateTime64:
+            column_type = ColumnType::MYSQL_TYPE_DATETIME;
+            flags = ColumnDefinitionFlags::BINARY_FLAG;
+            break;
+        case TypeIndex::Decimal32:
+        case TypeIndex::Decimal64:
+            column_type = ColumnType::MYSQL_TYPE_DECIMAL;
+            flags = ColumnDefinitionFlags::BINARY_FLAG;
+            break;
+        case TypeIndex::Decimal128: {
+            // MySQL Decimal has max 65 precision and 30 scale
+            const auto & type = assert_cast<const DataTypeDecimal128 &>(*data_type);
+            if (type.getPrecision() > 65 || type.getScale() > 30)
+            {
+                column_type = ColumnType::MYSQL_TYPE_STRING;
+                charset = CharacterSet::utf8_general_ci;
+            }
+            else
+            {
+                column_type = ColumnType::MYSQL_TYPE_DECIMAL;
+                flags = ColumnDefinitionFlags::BINARY_FLAG;
+            }
             break;
+        }
```

Comment:
The reader wonders why `TypeIndex::Decimal256` is not handled. According to [CH docs](https://clickhouse.com/docs/en/sql-reference/data-types/decimal), the minimum scale is 39 and therefore outside of MySQL's accepted range (l. 214).